### PR TITLE
Fix Target Type Validation

### LINKED ISSUE #2003: [BUG] Non-nullable are not valid InvalidOperationException is thrown when using VariableMultiValueConverter
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Did you read the "Reporting a bug" section on Contributing file?

- [X] I have read the "Reporting a bug" section on Contributing file: https://github.com/CommunityToolkit/Maui/blob/main/CONTRIBUTING.md#reporting-a-bug

### Current Behavior

System.InvalidOperationException: 'Non-nullable are not valid'

### Expected Behavior

MultiBinding works as it used to in prior versions.

### Steps To Reproduce

1- Give MainPage a reference name
     `x:Name="ThisPage"`

2- Copy and paste the following code in MainPage:

```
    <ContentPage.Resources>
        <toolkit:VariableMultiValueConverter ConditionType="All" x:Key="VariableMultiValueConverterAll" />
        <toolkit:InvertedBoolConverter x:Key="InvertedBoolConverter" />
    </ContentPage.Resources>

    <ScrollView>
        <VerticalStackLayout
            Padding="30,0"
            Spacing="25">
            <VerticalStackLayout.IsEnabled>
                <MultiBinding Converter="{StaticResource VariableMultiValueConverterAll}">
                    <Binding Path="IsEnabled" Source="{x:Reference ThisPage}" Converter="{StaticResource InvertedBoolConverter}" />
                    <Binding Path="IsVisible" Source="{x:Reference ThisPage}" Converter="{StaticResource InvertedBoolConverter}" />
                </MultiBinding>
            </VerticalStackLayout.IsEnabled>
        </VerticalStackLayout>
    </ScrollView>
```
3- Make sure CommunityToolkit.Maui 9.0.2 is installed
4- Non-nullable are not valid exception is thrown
5- Revert back to CommunityToolkit.Maui 9.0.1 to avoid this issue

Non-nullable2
 

### Link to public reproduction project repository

https://github.com/usefulBeeing/MauiNonNullableError

### Environment

```markdown
- .NET MAUI CommunityToolkit: 9.0.2
- OS: Windows 10.0.22631 Build 22631
- .NET MAUI: 8.0.61
```

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by brminnick**
### Description of Change ###

This fix updates the logic for `IsValidTargetType(Type, bool)` to use the `targetType` parameter passed into the method when verifying the nullable value types.

 ### Linked Issues ###
 

 - Fixes #2003 

 ### PR Checklist ###
 
 - [x] Has a linked Issue, and the Issue has been `approved`(bug) or `Championed` (feature/proposal)
 - [x] Has tests (if omitted, state reason in description)
 - [x] Rebased on top of `main` at time of PR
 - [x] Changes adhere to [coding standard](https://github.com/CommunityToolkit/Maui/blob/main/CONTRIBUTING.md#contributing-code---best-practices)

 ### Additional information ###

 I _think_ this is the fix. But I'll defer to @bijington as the expert here since he added the nullable value type check in this PR: https://github.com/CommunityToolkit/Maui/pull/1926

The exception being thrown in https://github.com/CommunityToolkit/Maui/issues/2003 was caused by using MultiBindings where `TTo` may not match `Type targetType`.

### CODE REVIEW

File: src/CommunityToolkit.Maui/Converters/VariableMultiValueConverter.shared.cs [hunk 1]
```diff
@@ -8,7 +8,7 @@ public enum MultiBindingCondition
 {
 	/// <summary>None of the values should be true.</summary>
 	None,
-	/// <summary>All of the values should be true.</summary>
+	/// <summary>All the values should be true.</summary>
```
Comment (brminnick) [inline src/CommunityToolkit.Maui/Converters/VariableMultiValueConverter.shared.cs:11]: Just cleaning up some grammar

File: src/CommunityToolkit.Maui/Converters/VariableMultiValueConverter.shared.cs [hunk 2]
```diff
@@ -66,7 +66,8 @@ public object Convert(object[]? values, Type targetType, object? parameter = nul
 			MultiBindingCondition.Exact => count == Count,
 			MultiBindingCondition.GreaterThan => count > Count,
 			MultiBindingCondition.LessThan => count < Count,
-			_ => count == boolValues.Length,
+			MultiBindingCondition.All => count == boolValues.Length,
+			_ => throw new NotSupportedException($"{ConditionType} not yet supported")
```
Comment (brminnick) [inline src/CommunityToolkit.Maui/Converters/VariableMultiValueConverter.shared.cs:70]: This doesn't change any logic. It just explicitly defines `MultiBindingCondition.All` and sets the default case to throw an exception. 

This default case will help us catch bugs in the future if/when we add another named constant to `MultiBindingCondition`.

**Review by bijington — APPROVED** [review]
Looks good to me @brminnick thanks for finishing this up. I just added the sample issue into the sample application as that was how I was testing this. I see you have a unit test covering it also though

**Comment by turdmaggot** [comment]
Hey, when will the new version of MAUI toolkit come out? We need this fix as soon as we can.

**Pull request merged**